lib/contextual_logger.rb

```diff
@@ -59,27 +59,27 @@ def current_context_for_thread
     end
 
     def debug(message = nil, context = {})
-      add(Logger::Severity::DEBUG, message || yield, **context)
+      add(Logger::Severity::DEBUG, message.nil? && block_given? ? yield : message, **context)
```

Comment:
Since we want to also transform the message to a string, is it better to move this to a more dry method?

```
def log_message(message, &blk)
  log_message= if message.nil? && blk.present?
    blk.call
  else
    message
  end
  log_message.inpect
end
```

Reply:
I was trying to avoid passing `&blk`--that costs more because the block has to be converted to a proc first. (That means building a closure for all the variables that were in scope at the time.)

Reply:
Ahh, okay. That optimization sounds like a good tradeoff for the non-dryness. Then all we need to do is convert all these to do the `inspect` at the end to make sure the message that results is a string.

Reply:
And note that we don't want to call `inspect` always. Just if the object is other than a `String` or `Exception`.

Reply:
^ I'll do this by calling a copy of `msg2str` inside the JSONify method.